Systems/c172p-engine.xml

```diff
@@ -0,0 +1,67 @@
+<?xml version="1.0"?>
+
+<!--
+
+  c172p damage effects for JSBSim.
+
+    Copyright (c) 2015 Gilberto Agostinho
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program; if not, write to the Free Software
+    Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+
+-->
+
+<system name="engine">
+
+    <channel name="oil">
+    
+        <fcs_function name="extra-weight-oil">
+            <function>
+                <product>
+                    <property>/engines/active-engine/oil-level</property>
+                    <value>1.7686</value>
+                </product>
+            </function>
+            <output>/fdm/jsbsim/inertia/pointmass-weight-lbs[16]</output>
+        </fcs_function>
+        
+    </channel>
+    
+    <channel name="carburetor">
+    
+        <fcs_function name="Carb heat on causes RPM drop on 160 HP engine">
+            <function>
+                <ifthen>
+                    <property>/controls/engines/current-engine/carb-heat</property>
```

Comment:
I think you need to use `/controls/anti-ice/engine[0]/carb-heat` here, and in the function below you use `/controls/anti-ice/engine[1]/carb-heat`.

The idea is that the user (pilot) controls `/controls/engines/current-engine/carb-heat`. The value of that property is then forwarded to either `/controls/anti-ice/engine[0]/carb-heat` or `/controls/anti-ice/engine[1]/carb-heat` depending on the selected engine. (So that JSBsim uses the correct value). Furthermore, for displaying and checklists you use `/controls/engines/current-engine/carb-heat` again (via the params in the model XML file). (So that animation system uses the correct value).

So for input (controls) and output (display) we use `/controls/engines/current-engine/carb-heat`, but for processing (JSBsim) we use `/controls/anti-ice/engine[0]/carb-heat` or `/controls/anti-ice/engine[1]/carb-heat`.